Regression when using `operator""s`

A regression was introduced in https://github.com/include-what-you-use/include-what-you-use/pull/2014, where IWYU incorrectly resolves `operator""s` provided by <chrono>.

Consider the following MRE:
```c++
#include <chrono>
using namespace std::chrono_literals;
int main() {
    auto a = 10s;
    return 0;
}
```

At commit f6400c725f0de1fabd6edb6bacaae65ee52cc746 (prior to the PR changes), IWYU evaluates the source file correctly:
```
$ include-what-you-use test.cpp 

(test.cpp has correct #includes/fwd-decls)
```

However, starting with commit ae18593bd2f3d4b16d8e9921aa95af0afb368490, IWYU fails:
```
$ include-what-you-use test.cpp 

test.cpp should add these lines:
#include <string>  // for operator""s

test.cpp should remove these lines:
- #include <chrono>  // lines 1-1

The full include-list for test.cpp:
#include <string>  // for operator""s
---
```

---

From https://en.cppreference.com/cpp/chrono/operator%22%22s:
> `std::string` also defines `operator""s`, to represent literal objects of type `std::string`, but it is a string literal: `10s` is ten seconds, but `"10"s` is a two-character string.

Looks like IWYU finds the right declaration when encountering the use:
```
/usr/include/c++/14/bits/chrono.h:584:3: (7 in tpl) [ RecordTypeLoc ] duration<long>
(Replaying full-use information from the cache for std::chrono::duration)
/usr/include/c++/14/bits/chrono.h:584:11: (7 in tpl) [ FunctionProtoTypeLoc ] void (void)
<invalid loc>: (8 in tpl) [ BuiltinTypeLoc ] void
Marked full-info use of decl 0x55604224cb18 std::operator""s() (from /usr/include/c++/14/bits/chrono.h:1393:7) at local-repros/2057.cc:4:14
/usr/include/c++/14/bits/chrono.h:1393:7: (0 in uninstantiated tpl) [ FunctionDecl ] 0x55604224cb18 constexpr chrono::seconds operator""s()
```

But then the symbol mapping doesn't have any overload information, and hits the parameterless overload for `std::string`:
```
--- Calculating IWYU violations for local-repros/2057.cc ---
Ignoring use of std::chrono::duration::~duration<:0, :1>() (local-repros/2057.cc:4:14): member of class
Mapped /usr/include/c++/14/bits/chrono.h to <string> for std::operator""s() (only candidate)
local-repros/2057.cc:4:14: warning: std::operator""s() is defined in <string>, which isn't directly #included.
```

--- a/mapgen/iwyu-mapgen-std-symbol.py
+++ b/mapgen/iwyu-mapgen-std-symbol.py
@@ -161,6 +161,10 @@ HANDWRITTEN_MAPPING = (
       'ostream'),
     ('std::print', 'print'),
     ('std::println', 'print'),
+# libstdc++ defines a literal operator for seconds as a numeric literal operator
+# template despite the <chrono> header synopsis. This may be standardized later.
+# See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85821.
+    ('std::operator""s()', 'chrono'),
 )
 
 def contains_unspec_alias(inp):

--- a/std_symbol_map.inc
+++ b/std_symbol_map.inc
@@ -1851,6 +1851,7 @@
 { "std::operator\"\"ns(long double)", UseKind::Full, "<chrono>", kPublic },
 { "std::operator\"\"ns(unsigned long long)", UseKind::Full, "<chrono>", kPublic },
 { "std::operator\"\"s", UseKind::Full, "<string>", kPublic },
+{ "std::operator\"\"s()", UseKind::Full, "<chrono>", kPublic },
 { "std::operator\"\"s(long double)", UseKind::Full, "<chrono>", kPublic },
 { "std::operator\"\"s(unsigned long long)", UseKind::Full, "<chrono>", kPublic },
 { "std::operator\"\"sv", UseKind::Full, "<string_view>", kPublic },

--- a/tests/cxx/std_symbol_mapping-direct.h
+++ b/tests/cxx/std_symbol_mapping-direct.h
@@ -86,4 +86,24 @@ using wspanstream = basic_spanstream<wchar_t>;
 template <typename T, typename = char_traits<T>>
 class basic_fstream;
 using fstream = basic_fstream<char>;
+
+using intmax_t = long long;
+
+template <intmax_t, intmax_t = 1>
+class ratio;
+
+namespace chrono {
+template <typename, typename = ratio<1>>
+class duration {};
+
+using seconds = duration<long long>;
+}  // namespace chrono
+
+inline namespace literals {
+inline namespace chrono_literals {
+// This approximately resembles how operator""s is defined in libstdc++.
+template <char...>
+chrono::seconds operator""s();
+}  // namespace chrono_literals
+}  // namespace literals
 }

--- a/tests/cxx/std_symbol_mapping.cc
+++ b/tests/cxx/std_symbol_mapping.cc
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-// IWYU_ARGS: -I .
+// IWYU_ARGS: -I . -Wno-user-defined-literals
 
 // Tests the internal IWYU standard library symbol mapping. In particular, tests
 // that the mapping works with types in the fwd-decl context.
@@ -84,6 +84,12 @@ void Fn() {
   std::basic_spanstream<char>* ss;
   // IWYU: std::basic_spanstream is...*<spanstream>
   (void)sizeof(*ss);
+
+  {
+    using namespace std::chrono_literals;
+    // IWYU: operator""s({{.*}}) is...*<chrono>
+    (void)10s;
+  }
 }
 
 /**** IWYU_SUMMARY
@@ -91,6 +97,7 @@ void Fn() {
 tests/cxx/std_symbol_mapping.cc should add these lines:
 #include <algorithm>
 #include <array>
+#include <chrono>
 #include <cmath>
 #include <functional>
 #include <iosfwd>
@@ -104,6 +111,7 @@ tests/cxx/std_symbol_mapping.cc should remove these lines:
 The full include-list for tests/cxx/std_symbol_mapping.cc:
 #include <algorithm>  // for move
 #include <array>  // for array, get, operator==
+#include <chrono>  // for operator""s
 #include <cmath>  // for pow
 #include <functional>  // for function, swap
 #include <iosfwd>  // for basic_ostream (ptr only), char_traits (ptr only)

--- a/tests/cxx/std_symbol_mapping_canonical.cc
+++ b/tests/cxx/std_symbol_mapping_canonical.cc
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-// IWYU_ARGS: -I .
+// IWYU_ARGS: -I . -Wno-user-defined-literals
 
 // Tests that symbols from std are by default attributed to their canonical
 // headers (<iosfwd> for std::fstream in a forward-declarable context). For this

--- a/tests/cxx/std_typedefs.cc
+++ b/tests/cxx/std_typedefs.cc
@@ -7,7 +7,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-// IWYU_ARGS: -I .
+// IWYU_ARGS: -I . -Wno-user-defined-literals
 
 // Tests handling typedefs provided by more than one standard library header.
 // Because wspanstream is used in the context requiring complete type info, it

